Content.Server/Radio/EntitySystems/MessagesServerSystem.cs

```diff
@@ -0,0 +1,129 @@
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Content.Server.CartridgeLoader.Cartridges;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Robust.Shared.Timing;
+using Content.Shared.CartridgeLoader;
+using Content.Server.CartridgeLoader;
+using Content.Server.Radio.Components;
+using Robust.Shared.Containers;
+using Content.Server.Power.Components;
+using Content.Server.Power.EntitySystems;
+using System.Linq;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Components;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Shared.DeviceNetwork;
+
+
+namespace Content.Server.Radio.EntitySystems;
+
+public sealed class MessagesServerSystem : EntitySystem
+{
+
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly MessagesCartridgeSystem _messagesCartridgeSystem = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+    [Dependency] private readonly SingletonDeviceNetServerSystem _singletonServerSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<MessagesServerComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+    }
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        var serverQuery = EntityQueryEnumerator<MessagesServerComponent>();
+        while (serverQuery.MoveNext(out var uid, out var server))
+        {
+            if (!_singletonServerSystem.IsActiveServer(uid))
+                continue;
+            if (server.NextUpdate <= _gameTiming.CurTime)
+            {
+                server.NextUpdate += server.UpdateDelay;
+
+                Update(uid, server);
+            }
+        }
+    }
+
```

Comment:
Why does a messaging server need to poll every second? Surely it can just subscribe to messages that come in instead?

Reply:
It's supposed to update every 10 seconds to broadcast a message asking for names of users on the network so it can send the list of other users to the clients. I thought that would be better than checking names on all carts every time someone opens the program. But, now that I think about it, that still seems like something to be gotten when needed. I'll update to that.
It also used the update function to sync with other servers, but I think I can just move that to the SendMessage function.